src/meter/__utils.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * The candidate optimum point is the midpoint between the minimum value and
+ * the maximum value.
+ *
+ * @see https://html.spec.whatwg.org/multipage/form-elements.html#the-meter-element:attr-meter-high-8:~:text=boundary.-,The%20optimum%20point
+ */
+export function getDefaultOptimumValue(min: number, max: number) {
+  return max < min ? min : min + (max - min) / 2;
+}
+
+/**
+ * Handle Inequalities with received values
+ *
+ * minimum ≤ value ≤ maximum
+ * minimum ≤ low ≤ maximum (if low is specified)
+ * minimum ≤ high ≤ maximum (if high is specified)
+ * minimum ≤ optimum ≤ maximum (if optimum is specified)
+ *
+ * @see https://html.spec.whatwg.org/multipage/form-elements.html#the-meter-element:attr-meter-max-3:~:text=following%20inequalities%20must%20hold
+ */
+export function clamp(value: number, min: number, max: number) {
+  if (value == null) return 0;
+
+  return Math.min(Math.max(value, min), max);
+}
+
+type CalculateStatusProps = {
+  value: number;
+  optimum: number;
+  min: number;
+  max: number;
+  low: number;
+  high: number;
+};
+
+export const calculateStatus = (props: CalculateStatusProps) => {
```

Comment:
Yup, looks good, i was also considering to extract the whole logic, sine it will help us unit test this part, this part really needs proper testing.

Reply:
Got it !!